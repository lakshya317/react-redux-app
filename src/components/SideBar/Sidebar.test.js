import { render } from '@testing-library/react';
import Sidebar from './Sidebar';
import { MemoryRouter } from 'react-router-dom';

describe('Sidebar Tests', () => {

    test('User Card Snapshot Test', () => {
        const sideBar = render(
            <MemoryRouter>
                <Sidebar/>
            </MemoryRouter>
        );

        expect(sideBar).toMatchSnapshot()
    });

    test('Sidebar text and icon verification', () => {
        const sideBar = render(
            <MemoryRouter>
                <Sidebar/>
            </MemoryRouter>
        );

        sideBar.getByText("Home");
        sideBar.getByText("Users");
        sideBar.getByAltText("Home");
        sideBar.getByAltText("Users");
    })
})