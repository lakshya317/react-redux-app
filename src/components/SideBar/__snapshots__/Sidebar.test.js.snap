// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Tests User Card Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sidebar-container"
      >
        <div
          class="sidebar-item"
        >
          <a
            class="menu-link"
            href="/home"
            style="text-decoration: none;"
          >
            <img
              alt="Home"
              class="menu-icon"
              src="home.png"
            />
            <div
              class="menu-title"
            >
              Home
            </div>
          </a>
        </div>
        <div
          class="sidebar-item"
        >
          <a
            class="menu-link"
            href="/users"
            style="text-decoration: none;"
          >
            <img
              alt="Users"
              class="menu-icon"
              src="users.png"
            />
            <div
              class="menu-title"
            >
              Users
            </div>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sidebar-container"
    >
      <div
        class="sidebar-item"
      >
        <a
          class="menu-link"
          href="/home"
          style="text-decoration: none;"
        >
          <img
            alt="Home"
            class="menu-icon"
            src="home.png"
          />
          <div
            class="menu-title"
          >
            Home
          </div>
        </a>
      </div>
      <div
        class="sidebar-item"
      >
        <a
          class="menu-link"
          href="/users"
          style="text-decoration: none;"
        >
          <img
            alt="Users"
            class="menu-icon"
            src="users.png"
          />
          <div
            class="menu-title"
          >
            Users
          </div>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
